utils/sl/check_script.py

```diff
@@ -143,43 +135,37 @@ def write_variables(outfile, prefix, variables):
     @staticmethod
     def write_header(outfile):
         """write the header"""
-        outfile.write("""# pylint: disable=wildcard-import
-# pylint: disable=import-outside-toplevel
-from __future__ import print_function
-from SCons.Script import *
-from SCons.Variables import *
-# pylint: enable=wildcard-import
-# pylint: enable=import-outside-toplevel\n""")
-        return 7
+        outfile.write("""# Autoinserted wrapper header
+from SCons.Script import * # pylint: disable=import-outside-toplevel,wildcard-import
+from SCons.Variables import * # pylint: disable=import-outside-toplevel,wildcard-import\n""")
+        return 3
+
+    def convert_line(self, line):
+        """Convert from a line number in the report to a line number in the input file"""
+        return self.line_map[line]
 
     def fix_log(self, log_file, fname):
         """Get the line number"""
         os.unlink("script")
         log_file.seek(0)
-        try:
-            output = tempfile.TemporaryFile(mode='w+', encoding='utf-8')
-        except TypeError:
-            output = tempfile.TemporaryFile()
+        output = tempfile.TemporaryFile(mode='w+', encoding='utf-8')
         for line in log_file.readlines():
             match = re.search(r":(\d+):", line)
             if match:
                 lineno = int(match.group(1))
-                if int(lineno) in self.line_map.keys():
-                    line = line.replace(str(lineno),
-                                        str(self.line_map[lineno]),
-                                        1)
+                if int(lineno) in self.line_map:
+                    line = line.replace(str(lineno), str(self.line_map[lineno]), 1)
             match = re.search("^(.*)Module script(.*)$", line)
             if match:
-                line = "%sModule %s%s\n" % (match.group(1),
-                                            fname,
-                                            match.group(2))
+                line = "%sModule %s%s\n" % (match.group(1), fname, match.group(2))
```

Comment:
(pylint-consider-using-f-string) Formatting a regular string which could be a f-string